src/Microsoft.SqlTools.ServiceLayer/Profiler/ProfilerService.cs

```diff
@@ -205,8 +205,27 @@ internal async Task HandleStopProfilingRequest(StopProfilingParams parameters, R
 
                 if (session != null)
                 {
-                    session.XEventSession.Stop();
-                    await requestContext.SendResult(new StopProfilingResult{});
+                    // Occasionally we might see the InvalidOperationException due to a read is 
+                    // in progress, add the following retry logic will solve the problem.
+                    int remainingAttampts = 3;
+                    while (true)
+                    {
+                        try
+                        {
+                            session.XEventSession.Stop();
+                            await requestContext.SendResult(new StopProfilingResult { });
+                            break;
+                        }
+                        catch(InvalidOperationException)
+                        {
+                            remainingAttampts--;
+                            if (remainingAttampts == 0)
+                            {
+                                throw;
+                            }
+                            Thread.Sleep(500);
```

Comment:
this request handler should probably be moved into a different thread.  I think as the dispatcher is currently written this will block other events while the retries are going on.